code/modules/mechs/equipment/combat.dm

```diff
@@ -1034,10 +1037,200 @@
+		if(ishuman(target))
+			var/mob/living/carbon/human/targ = target
+			if(targ.stats.getStat(STAT_VIG) > STAT_LEVEL_EXPERT)
+				targ.visible_message(SPAN_DANGER("[targ] dodges the [holding] slam!"), "You dodge [loc]'s [holding] slam!", "You hear a woosh.", 6)
+				return
+			targ.visible_message(SPAN_DANGER("[targ] gets slammed by [src]'s [holding]!"), SPAN_NOTICE("You get slammed by [src]'s [holding]!"), "You hear something soft hit a metal plate!", 6)
+			targ.Weaken(1)
+			targ.throw_at(get_turf_away_from_target_complex(target,user,3), 5, 1, loc)
+			targ.damage_through_armor(20, BRUTE, BP_CHEST, ARMOR_MELEE, 1, src, FALSE, FALSE, 1)
+		else
+			target.visible_message(SPAN_DANGER("[target] gets slammed by [src]'s [holding]!"), SPAN_NOTICE("You get slammed by [src]'s [holding]!"), "You hear something soft hit a metal plate!", 6)
+			target.Weaken(1)
+			target.throw_at(get_turf_away_from_target_complex(target,user,3), 3, 1, loc)
+			target.damage_through_armor(20, BRUTE, BP_CHEST, ARMOR_MELEE, 2, src, FALSE, FALSE, 1)
+
+
+/obj/item/mech_equipment/mounted_system/mace/get_overlay_state()
+	return "[icon_state]_[active ? "on" : "off"]"
 
+/obj/item/tool/hammer/mace/mech
+	name = "mace head"
+	desc = "What are you standing around staring at this for? You shouldn't be seeing this..."
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "mace"
+	item_state = "mace"
+	matter = list(MATERIAL_PLASTEEL = 15, MATERIAL_PLASTIC = 5)
+	w_class = ITEM_SIZE_BULKY
+	worksound = WORKSOUND_HAMMER
+	wielded = TRUE
+	canremove = FALSE
+	// Its Big
+	armor_divisor = ARMOR_PEN_DEEP
+	tool_qualities = list(QUALITY_HAMMERING = 45)
+	// its mech sized!!!!!
+	structure_damage_factor = STRUCTURE_DAMAGE_DESTRUCTIVE
+	spawn_blacklisted = TRUE
+	force= WEAPON_FORCE_BRUTAL
+	force_wielded_multiplier = 1.5
+
+/obj/item/tool/hammer/mace/mech/attack_self(mob/user)
+	. = ..()
+	return TRUE
+
+/obj/item/mech_equipment/mounted_system/bfg
+	name = "mounted BFG"
+	icon_state = "plasmabfg"
+	holding_type = /obj/item/gun/energy/plasma_mech
+	restricted_software = list(MECH_SOFTWARE_ADVWEAPONS)
+	restricted_hardpoints = list(HARDPOINT_LEFT_HAND, HARDPOINT_RIGHT_HAND)
+	origin_tech = list(TECH_MATERIAL = 4, TECH_PLASMA = 4, TECH_ENGINEERING = 6, TECH_COMBAT = 3)
+	matter = list(MATERIAL_PLASTEEL = 20, MATERIAL_STEEL = 10, MATERIAL_SILVER = 10)
+	spawn_tags = SPAWN_MECH_QUIPMENT
+	spawn_blacklisted = FALSE
+	rarity_value = 60
 
+/obj/item/gun/energy/plasma_mech
+	name = "mounted BFG"
+	desc = "A large, bulky weapon that fires a massive energy blast. It's a bit unwieldy, but it packs a punch."
+	safety = FALSE
+	spawn_tags = null
+	spawn_blacklisted = TRUE
+	use_external_power = TRUE
+	self_recharge = TRUE
+	restrict_safety = TRUE
+	twohanded = FALSE
+	charge_cost = MECH_WEAPON_POWER_COST * 5
+	projectile_type = /obj/item/projectile/plasma/aoe/heat/strong/mech
+	fire_sound='sound/weapons/energy/melt.ogg'
+	burst = 1
+	fire_delay = 120
```

Comment:
With this delay don't be afraid to make the projectile pack a punch. Potentially a greater area of effect, but most importantly just more heat damage in general.

Reply:
It's not supposed to have a 12 second delay, but rather be charged up to fire.